pkg/apis/multicluster/v1alpha1/gameserverallocationpolicy.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GameServerAllocationPolicySpec defines the desired state of GameServerAllocationPolicy
+type GameServerAllocationPolicySpec struct {
+	// +kubebuilder:validation:Minimum=0
+	Priority int `json:"priority"`
+	// +kubebuilder:validation:Minimum=0
+	Weight         int                   `json:"weight"`
+	ConnectionInfo ClusterConnectionInfo `json:"connectionInfo,omitempty"`
+}
+
+// ClusterConnectionInfo defines the connection information for a cluster
+type ClusterConnectionInfo struct {
+	ClusterName             string        `json:"clusterName"`
+	APIServerEndpoint       string        `json:"apiServerEndpoint"`
+	AllocationServiceSecret corev1.Secret `json:"allocationServiceSecret"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// GameServerAllocationPolicy is the Schema for the gameserverallocationpolicies API
+// +k8s:openapi-gen=true
+type GameServerAllocationPolicy struct {
```

Comment:
Super mega minor nit - we have tended to put the root declaration and the list at the top.

Only move if you want to.

Reply:
I'll keep it as is. It is generated by kubebuilder.